### PR TITLE
Add Swift Package Manager support

### PR DESCRIPTION
**Pull request by charafau**
Swift Package Manager is supported by flutter so I updated this plugin to support this as well - based on https://docs.flutter.dev/packages-and-plugins/swift-package-manager/for-plugin-authors

- Swift package manager support
- bump min ios version to 12
- bump min macos version to 10.14
- bump android min sdk to 21 in example
- update pigeon to 21.2 (bug fix for spm)
- update git ignore

### CODE REVIEW

**Comment by diegotori** [comment]
@charafau also, please revert the `pigeon` related changes, since they'll be addressed in another PR, which will target `wakelock_plus_platform_interface`.

File: wakelock_plus/ios/wakelock_plus/Sources/wakelock_plus/WakelockPlusPlugin.m [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#import "WakelockPlusPlugin.h"
-#import "messages.g.h"
-#import "UIApplication+idleTimerLock.h"
+#import "./include/wakelock_plus/WakelockPlusPlugin.h"
+#import "./include/wakelock_plus/messages.g.h"
+#import "./include/wakelock_plus/UIApplication+idleTimerLock.h"
```
Comment (diegotori) [inline wakelock_plus/ios/wakelock_plus/Sources/wakelock_plus/WakelockPlusPlugin.m:3]: Why are we changing the source paths? Can this be accomplished without having to do this?

**Review by diegotori — CHANGES_REQUESTED** [review]

**Comment by diegotori** [comment]
@charafau I'm gonna submit a PR shortly that upgrades `pigeon` to `21.2` so that `wakelock_plus_platform_interface` is upgraded before your changes.

**Comment by diegotori** [comment]
@charafau because this requires switch Flutter to the `main` channel, I'm gonna hold off on approving this PR.

Once the Flutter team releases SPM support in the `stable` channel, then I'll revisit your PR.

Once again, thanks for your contribution.

File: wakelock_plus/ios/wakelock_plus/Sources/wakelock_plus/WakelockPlusPlugin.m (hunk 1 above)
Comment (charafau) [reply wakelock_plus/ios/wakelock_plus/Sources/wakelock_plus/WakelockPlusPlugin.m:3]: According to the docs, public headers should be added to `include` path

**Comment by charafau** [comment]
> Once the Flutter team releases SPM support in the `stable` channel, then I'll revisit your PR.

Ok, I think we might prepare support for SPM before that though. Let me know, I will probably add support for more community packages. Closing this for now then

File: wakelock_plus/ios/wakelock_plus/Sources/wakelock_plus/WakelockPlusPlugin.m (hunk 1 above)
Comment (diegotori) [reply wakelock_plus/ios/wakelock_plus/Sources/wakelock_plus/WakelockPlusPlugin.m:3]: Understood. Nevertheless, I'm gonna hold off on this PR until the Flutter team finalizes SPM support.

**Comment by diegotori** [comment]
> > Once the Flutter team releases SPM support in the `stable` channel, then I'll revisit your PR.
> 
> Ok, I think we might prepare support for SPM before that though. Let me know, I will probably add support for more community packages. Closing this for now then

Totally get where you're coming from. However, I also have to weigh your changes against something that is currently non-final and subject to change. 

Otherwise, we would be doing twice the work if the Flutter team changes the SPM integration between now and when it goes stable.

**Comment by SputNikPlop** [comment]
Another idea to consider would be having a beta branch since this will reach the beta branch of flutter soon

**Pull request closed without merging**